src/battle.ts

```diff
@@ -2966,14 +2973,14 @@ class Battle {
 		pokemonid = parsedPokemonid;
 
 		const searchid = `${pokemonid}|${details}`;
-		const side = this.sides[siden];
+		const side = this.gameType === 'multi' ? this.getSide(`p${siden + 1}`) : this.sides[siden];
 
 		// search inactive revealed pokemon
 		for (let i = 0; i < side.pokemon.length; i++) {
 			let pokemon = side.pokemon[i];
 			if (pokemon.fainted) continue;
 			// already active, can't be switching in
-			if (side.active.includes(pokemon)) continue;
+			if (side.active.includes(pokemon) && side.ally) continue;
```

Comment:
This shouldn't matter. Active Pokémon shouldn't be able to switch in.

Reply:
I honestly have no idea why that's there. Git blame isn't any help, either, so I guess I'll try taking it out and see what happens.